ppr-ui/src/components/common/UpdatedBadge.vue

```diff
@@ -0,0 +1,32 @@
+<template>
+  <v-chip
+    v-if="hasChanges"
+    id="updated-badge-component"
+    xSmall
+    variant="flat"
+    color="primary"
+    :data-test-id="`${action}-badge`"
+  >
+    <b>{{ action }}</b>
+  </v-chip>
+</template>
+<script setup lang="ts">
+import { computed } from 'vue'
+import { BaseDataUnionIF } from '@/interfaces'
+import { deepChangesComparison } from '@/utils'
+
+const props = withDefaults(defineProps<{
+  action?: string,
+  baseline: BaseDataUnionIF,
+  currentState: BaseDataUnionIF
+}>(), {
+  action: 'CORRECTED',
+  baseline: null,
+  currentState: null
+})
```

Comment:
@dimak1 Small heads up here, when deconstructing props to apply defaults as in MhrTransportPermit line 258, it has reactivity implications. I followed that same approach initially and was losing reactivity on those props. 

<img width="776" alt="Screenshot 2024-02-16 at 11 18 39 AM" src="https://github.com/bcgov/ppr/assets/53542131/8e793b51-0762-4519-a100-ca1f97e2c42d">

Reply:
Might just be when trying to type our props, as it states in the documentation there. 
Either way, not sure if it was an issue in the TransportPermit, just something to be aware of!